File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        front = 0
        last = len(height)-1
        area = (last-front)*min(height[front],height[last])
        while front!=last:
            if height[front]<height[last]:
                front+=1
            else:
                last-=1
            if area < ((last-front)*min(height[front],height[last])):
                area = ((last-front)*min(height[front],height[last]))
        return area

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int start = 0;
        int end = height.length-1;
        int ans = (end-start)*Math.min(height[start],height[end]);

        while(end > start){
            if(height[start]>=height[end]){
                end--;
            }else{
                start++;
            }
            ans = Math.max(ans, (end-start)*Math.min(height[start],height[end]));
        }
        return ans;
    }
}